src/test/java/rx/internal/operators/OnSubscribeFromArrayTest.java

```diff
@@ -64,4 +65,15 @@ public void backpressure() {
         ts.assertCompleted();
     }
 
+    @Test
+    public void empty() {
+        Assert.assertSame(Observable.empty(), Observable.from(new Object[0]));
+    }
+
+    @Test
+    public void just() {
+        Observable<Integer> source = Observable.from(new Integer[] { 1 });
+        Assert.assertTrue(source.getClass().toString(), source instanceof ScalarSynchronousObservable);
```

Comment:
No, this tests that the scalar optimization is applied by checking the return type.